usaspending_api/references/management/commands/update_location_usage_flags.py

```diff
@@ -0,0 +1,43 @@
+from django.core.management.base import BaseCommand, CommandError
+from django.db.models import Count, Case, F, Value, When, Q
+from usaspending_api.references.models import Location
+import logging
+import django
+
+
+class Command(BaseCommand):
+    help = "Updates Location object usage flags based upon their current usage."
+    logger = logging.getLogger('console')
+
+    def handle(self, *args, **options):
+        annotated_set = Location.objects.annotate(legalentity_count=Count('legalentity'))
+        annotated_set = annotated_set.annotate(award_count=Count('award'))
+        annotated_set = annotated_set.annotate(procurement_count=Count('procurement'))
+        annotated_set = annotated_set.annotate(assistance_count=Count('financialassistanceaward'))
+
+        # Locations have a recipient flag of true if the number of legal entities using
+        # them is greater than or equal to 1
+        # We could combine all of the following procedures into one line, but that would
+        # greatly reduce readability
```

Comment:
Thinking about it more, I believe we could simplify things further by doing something like:

```python
# get a queryset of location records with ids that are present in the legal entity location_id field
le_locations = Location.objects.filter(location_id__in=LegalEntity.objects.values('location'))
# update the recipient flag on the locations in that queryset
le_locations.update(recipient_flag = True)

# repeat for place of performance
```